core/server/models/post.js

```diff
@@ -302,6 +302,17 @@ Post = ghostBookshelf.Model.extend({
         return sequence(ops);
     },
 
+    onValidate: function onValidate() {
+        // CASE: the client might send empty image properties with "" instead of setting them to null.
+        // This can cause GQL to fail. We therefore enforce 'null' for empty image properties.
+        // See https://github.com/TryGhost/GQL/issues/24
+        var affectedProps = ['feature_image', 'og_image', 'twitter_image'];
+
+        ghostBookshelf.Model.prototype.setEmptyValuesToNull.call(this, affectedProps);
```

Comment:
I would suggest to move the logic into the parent/base fn. We currently have to override the base fn three times and it's doing the same.

```
onValidate: function onValidate(model, columns, options) {
    this.careAboutEmptyString(...);
    return validation.validateSchema(this.tableName, this, options);
}

careAboutEmptyString: fn
  if (!this.emptyStringFields)
     return;

  const properties = this.emptyStringFields();
  ...
```

e.g. in Post model

```
emptyStringFields: fn
   return ['...'];
```

Naming of functions is just an example and not good enough.

Reply:
This has also the advantage, that you can easily add the functionality for other models in the future (without overriding the validate fn)

Reply:
Makes sense to me!